docs/DeveloperGuide.md

```diff
@@ -1,33 +1,98 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+Ideas for this project partly comes from NUSmods at https://nusmods.com/
+
+API and library used includes the following:
+- [NUSMods API](https://api.nusmods.com/v2/) - To extract information about modules in NUS
+- [FasterXML](https://github.com/FasterXML/jackson) - Helps parse json files from the NUSMods API
 
+## Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
 
+
+#### Architecture 
+<img src="images/architectureDiagram.png" width="280" />
+
+
+The ***Architecture Diagram*** given above presents the high-level design of the product. The main components included are as follows:
+1. `Duke`: Initialised when launch and acts as the main logic component of the program.
+2. `UI`: Handles all user interaction, including printing responses and receiving user inputs.
+3. `Parser`: Parses main user commands and calls for appropriate functions.
+4. `commands`: Exists as a package which includes all command functions.
+5. `Timetable`: Contains all the data in the current session. (Data can only be accessed and manipulated by commands)
+6. `data`: Manages the saving and loading of the current session data.
+7. `Nusmods`: Handles all communication to the NUSMods API and extracts useful information that is used for the rest of 
+the program.
+
+<p style='text-align: justify;'>The design of the product focuses on low coupling and high cohesion. Following as closely to the Single Responsibility Principle (SRP) as possible, each major component explained above has unique roles which is not overlapped by another component. For example, printing of all responses go through the UI class, which handles all interactions with user, including the display of program responses.</p>
+
+#### Key Features
+##### Adding modules to current timetable
```

Comment:
Font size for this header is small compared to the description. Perhaps increase the font size to make it more noticeable.